### PR TITLE
Framework for Nebari deployment via pytest for extensive testing

### PR DESCRIPTION
**Pull request by aktech**
Part of fixing #1862 

This creates a framework to write tests that are designed to test things on Nebari deployed on cloud. At the moment it only deploys on DigitalOcean, which can be easily expanded to all other clouds.

It would be better to merge this small chunk and then work on other cloud providers, rather than a single PR handling all clouds making it difficult to review.

Testing on cloud would be simply adding the decorator `@on_cloud("cloud_name")`:

```python
@on_cloud("do")
def test_something_on_do_deployment(deploy):
	# create notebook
    # test dask
    # test jupyterhub
    # test conda store
```

Other Fixes:

- [x] Fixes fetching Kubernetes Version for DigitalOcean
- [x] Playwright cleanups 

Notes: 
- **More extensive testing like, running notebooks, dask, jupyterhub will be done in follow up PRs**
- Example run: https://github.com/nebari-dev/nebari/actions/runs/5620616744/job/15229991582
- Will also add pytest markers for clouds/testing type, etc in subsequent PRs

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

### CODE REVIEW

**Comment by iameskild** [comment]
Reminder for @costrouc and @iameskild to test this as soon as possible.

**Review by iameskild — COMMENTED** [review]
Thanks for the great work @aktech!

Given that this will be used to deploy cloud-based Nebari clusters, will this replace some of the manual setup we do for the integration tests? That's what I recall from our previous conversation with @costrouc. 

Lastly, I'm unfortunately unable to test this locally because Digital Ocean just dropped support for kubernetes 1.24...

File: .github/workflows/integration_test.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Integration Tests
+        run: |
+          pytest --version
+          pytest tests_integration/ -vvv -s
+        env:
+          NEBARI_K8S_VERSION: 1.24.13-do.0
+          DIGITALOCEAN_TOKEN: ${{ secrets.DIGITALOCEAN_TOKEN }}
+          SPACES_ACCESS_KEY_ID: ${{ secrets.SPACES_ACCESS_KEY_ID }}
+          SPACES_SECRET_ACCESS_KEY: ${{ secrets.SPACES_SECRET_ACCESS_KEY }}
+          CLOUDFLARE_TOKEN: ${{ secrets.CLOUDFLARE_TOKEN }}
```
[the remark this answers is not in the document]
Comment (aktech) [reply .github/workflows/integration_test.yaml:35]: Yes, absolutely. I tried to use it yesterday from the given workflow directly, but probably it didn't had enough permissions. I don't have access to repository secrets in vault. I have asked @costrouc 

Here is that failed run (due to vault secrets not having enough permissions(?)):
https://github.com/nebari-dev/nebari/actions/runs/5732309497/job/15535014488

File: src/_nebari/provider/cloud/digital_ocean.py [hunk 2]
```diff
@@ -44,6 +44,9 @@ def regions():
...
     """Return list of available kubernetes supported by cloud provider. Sorted from oldest to latest."""
     supported_kubernetes_versions = sorted(
-        [_["slug"] for _ in _kubernetes_options()["options"]["versions"]]
+        [_["slug"].split("-")[0] for _ in _kubernetes_options()["options"]["versions"]]
     )
-    return filter_by_highest_supported_k8s_version(supported_kubernetes_versions)
+    filtered_versions = filter_by_highest_supported_k8s_version(
+        supported_kubernetes_versions
+    )
+    return [f"{v}-do.0" for v in filtered_versions]
```
[the remark this answers is not in the document]
Comment (aktech) [reply src/_nebari/provider/cloud/digital_ocean.py:52]: Yes, seems like, from what I got from their API.

File: .github/workflows/integration_test.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (iameskild) [reply .github/workflows/integration_test.yaml:35]: Ah okay, when I set the permissions for the AWS credentials ([here](https://github.com/Quansight/quansight-cloud-infrastructure/blob/dd12ec8b9e5448a6ee3af281c5f9ed33f8319080/data/amazon-web-services.yaml#L69-L70)), I gave them the read-only permissions. We will need to elevate the them to `AdministratorAccess` for now.

**Comment by iameskild** [comment]
@aktech the one other thing that will likely be worthwhile adding is some sort of clean up script so in the event things are fully destroyed, we can still ensure the cloud resources are removed. We have a script for AWS (python and bash) but for the other cloud providers, we can either copy over what we have from the integration tests or write a python script that handles this for us.

File: .github/workflows/integration_test.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (iameskild) [reply .github/workflows/integration_test.yaml:35]: PR for this opened [here](https://github.com/Quansight/quansight-cloud-infrastructure/pull/129).

**Review by costrouc — APPROVED** [review]

**Pull request merged**